base/applications/charmap/charmap.c

```diff
@@ -312,6 +312,10 @@ ChangeView(HWND hWnd)
 #ifndef REMOVE_ADVANCED
     if (Settings.IsAdvancedView)
         Height += rcAdvanced.bottom;
+#else
+    /* The lack of advanced button leaves an empty gap at the bottom of the window.
+       Shrink the window height a bit here to accomodate for that lost control. */
+    Height -= 30;
```

Comment:
Can this be calculated based on the controls? Since the dialog layout is defined in the resources, a different size could be appropriate for different languages.
Something like `IDC_CHECK_ADVANCED.Bottom - IDC_COPY.Bottom`?

Reply:
Sort of.  The advanced dialog is removed from the control.  I can find the bottom of the IDC_COPY (by getting rcCharmap.bottom + IDC_COPY.Bottom) and then add some padding, but that padding is kind of a magic number (see new commit).  Perhaps that's good enough?  Let me know what you think.

Also worth noting that I now have two assignments to Height.  The original, and then the new calculation.  I left the original for now so that DeY is used, since the warning if it isn't used will trigger a compilation error.  Otherwise there will be a bunch of code to either eliminate DeY when compiling with REMOVE_ADVANCED, or I'll have to use it with another magic number.  Something like Height = rcCharmap.bottom + rcCopy.bottom + DeY - 29

Reply:
Seems good to me. Thanks!